test/integration/examples/apiserver_test.go

```diff
@@ -90,7 +90,7 @@ func TestAggregatedAPIServer(t *testing.T) {
 			"--kubeconfig", wardleToKASKubeConfigFile,
 		})
 		if err := wardleCmd.Execute(); err != nil {
-			t.Fatal(err)
+			t.Errorf("unexpected error: %v", err)
```

Comment:
could you explain this change?

Reply:
I got this: 
```
Errors from staticcheck:
test/integration/examples/apiserver_test.go:80:2: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)
test/integration/examples/apiserver_test.go:93:4: call to T.Fatalf
```
https://github.com/kubernetes/kubernetes/blob/master/test/integration/examples/apiserver_test.go#L78

This function is used by some other function so I assume changing this will not change the test behavior?

Reply:
having another look,
the correct fix in such a case is to capture the potential error from the goroutine and call t.Fatal in the main thread.

@kubernetes/sig-api-machinery-pr-reviews

Reply:
@neolit123 thanks for the advice. It makes sense to me. I think it is a better fix. Will update